src/checksums.rs

```diff
@@ -0,0 +1,27 @@
+//! # Implementation of the interface to custom checksum generator
+
+use cosmwasm_std::{Addr, HexBinary};
+use sha2::{Digest, Sha256};
+
+/// An interface to call custom checksum generator for contract's code.
+///
+/// This trait defines a method to calculate checksum based on
+/// the creator's address and a unique code identifier.
+pub trait ChecksumGenerator {
+    /// Calculates the checksum for a given contract's code creator
+    /// and code identifier. Returns a hexadecimal binary representation
+    /// of the calculated checksum. There are no assumptions about
+    /// the length of the calculated checksum.
+    fn checksum(&self, creator: Addr, code_id: u64) -> HexBinary;
```

Comment:
This interface works for our use case, but I wonder if we should pass something else in here. I guess it depends a bit on the use cases we imagine people use this for. If it is just hardcoding some hash for a contract, then this would be good enough, but if we expect that people need more info about the contract code, then maybe we should pass the WasmKeeper here? Not sure, just a question that popped up in my mind.

Reply:
Yes, I was also thinking about some more additional attributes that our users would need.
Passing `Api` gives nothing more as address validation.
Passing `Storage` is not helpful unless someone stores there something that would be used in this handler.
Passing `Wasm` does not help in any case while this function is called before anything is stored.
To be honest, I have now no reasonable idea what could be passed here now, what user would need.
I guess the best way is to wait until someone has a real-life use case and then change this API.

Reply:
Ah, you are right, the code is not stored yet when this is called. Then let's keep it like it is for now.